cli/azd/internal/repository/infra_confirm.go

```diff
@@ -167,10 +131,10 @@ func (i *Initializer) infraSpecFromDetect(
 	return spec, nil
 }
 
-func (i *Initializer) getPortByPrompt(ctx context.Context, promptMessage string) (int, error) {
+func promptPortNumber(console input.Console, ctx context.Context, promptMessage string) (int, error) {
```

Comment:
consider moving this functionality to the input.Console by using the `input.ConsoleOptions` where you could define the type Number and the range.  This would benefit the entire product and not only things around composibility

Reply:
I could be wrong, but I think you're asking for features of a UI dialog prompt library, such as the default `survey.Questions`: https://pkg.go.dev/github.com/AlecAivazis/survey/v2#section-readme.

`azd` had decidedly avoid this multi-prompt dialog pattern to gain control over each interaction. I do somewhat see over time, as we're building more complex prompts, the need for this. But this isn't a suggestion that we can land within a feature change itself; let's take a holistic consideration on this over time.